File: app/src/main/java/com/pixels/blockies/app/game/figures/FigureZRevert.java
package com.pixels.blockies.app.game.figures;

import com.pixels.blockies.app.game.Grid;

public class FigureZRevert extends AbstractFigure {

    private int[][][] figures = new int[][][]{
            {
                    {Grid.EMPTY, 6},
                    {6, 6},
                    {6, Grid.EMPTY}
            },
            {
                    {6, 6, Grid.EMPTY},
                    {Grid.EMPTY, 6, 6}
            }
    };

    @Override
    protected int getFigureCount() {
        return figures.length;
    }

    @Override
    public int[][] get() {
        return figures[getCurrentRotation()];
    }

    @Override
    public int[][] getNext() {
        return figures[getNextRotation()];
    }
}

File: app/src/main/java/com/pixels/blockies/game/game/figures/FigureZRevert.kt
package com.pixels.blockies.game.game.figures

import com.pixels.blockies.game.game.Grid

class FigureZRevert : AbstractFigure() {
    private val figures = arrayOf(
        arrayOf<IntArray?>(
            intArrayOf(Grid.Companion.EMPTY, 6),
            intArrayOf(6, 6),
            intArrayOf(6, Grid.Companion.EMPTY)
        ),
        arrayOf<IntArray?>(
            intArrayOf(6, 6, Grid.Companion.EMPTY),
            intArrayOf(Grid.Companion.EMPTY, 6, 6)
        )
    )

    override fun getFigureCount(): Int {
        return figures.size
    }

    override fun get(): Array<IntArray?> {
        return figures[currentRotation]
    }

    override fun getNext(): Array<IntArray?> {
        return figures[nextRotation]
    }
}